Quorum/apis/block_explorers/source_code.py

```diff
@@ -84,51 +71,7 @@ def get_functions(self) -> dict | None:
             (dict | None): Dictionary of functions or None if not found.
         """
         if self._parsed_contract:
-            return self._parsed_contract.functions
-        return None
-
-    def get_inherited_names(self) -> list | None:
-        """
-        Retrieves the inherited names from the Solidity contract.
-
-        Returns:
-            (list | None): List of inherited names or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.inherited_names
-        return None
-
-    def get_mappings(self) -> list | None:
-        """
-        Retrieves the mappings from the Solidity contract.
-
-        Returns:
-            (list | None): List of mappings or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.mappings
-        return None
-
-    def get_modifiers(self) -> list | None:
-        """
-        Retrieves the modifiers from the Solidity contract.
-
-        Returns:
-            (list | None): List of modifiers or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.modifiers
-        return None
-
-    def get_name(self) -> str | None:
-        """
-        Retrieves the contract name.
-
-        Returns:
-            (str | None): Contract name or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.name
+            return self.__extract_nodes(self._parsed_contract, ASTOption.FUNCTIONS)
```

Comment:
You can cache the result on the object as (I think) the source code will not change.

Reply:
Done